corehq/form_processor/submission_post.py

```diff
@@ -297,7 +297,7 @@ def run(self):
                     submission_type = 'normal'
                     try:
                         case_stock_result = self.process_xforms_for_cases(xforms, case_db)
-                    except (IllegalCaseId, UsesReferrals, MissingProductId,
+                    except (IllegalCaseId, UsesReferrals, MissingProductId, InvalidCaseId,
```

Comment:
How is `InvalidCaseId` different from `IllegalCaseId`? Could we use that instead of adding a new exception type and dependency of `casexml.apps.case` on `corehq.form_processor`?

Reply:
@millerdev `IllegalCaseId` appears to be used for multiple things including incorrect case references from the usages [usages](https://github.com/dimagi/commcare-hq/search?q=%22raise+IllegalCaseId%22) . So having a distinction of a clear exception about the case id itself would be nice, though IllegalCaseId might have been used in the same context as well of an invalid case id. Its not very clear. Though i am fine with using what is already present if having a distinct exception sounds an overkill.

If we still keep it, is `casexml.apps.case.exceptions` a better place for `InvalidCaseId`?

Reply:
+1 to reuse `IllegalCaseId` and just use the message to indicate the error